pkg/scheduler/algorithm/priorities/interpod_affinity.go

```diff
@@ -36,47 +38,68 @@ import (
+	namespaces  sets.String
+	selector    labels.Selector
+	weight      int32
+	topologyKey string
 }
 
-func (p *podAffinityPriorityMap) processTerm(term *v1.PodAffinityTerm, podDefiningAffinityTerm, podToCheck *v1.Pod, fixedNode *v1.Node, weight int64) error {
-	namespaces := priorityutil.GetNamespacesFromPodAffinityTerm(podDefiningAffinityTerm, term)
+func getProcessedWeightedAffinityTerm(pod *v1.Pod, term *v1.PodAffinityTerm, weight int32) (*processedWeightedAffinityTerm, error) {
+	namespaces := priorityutil.GetNamespacesFromPodAffinityTerm(pod, term)
 	selector, err := metav1.LabelSelectorAsSelector(term.LabelSelector)
 	if err != nil {
-		return err
+		return nil, err
+	}
+	return &processedWeightedAffinityTerm{namespaces: namespaces, selector: selector, topologyKey: term.TopologyKey, weight: weight}, nil
+}
+
+func getProcessedTerms(pod *v1.Pod, terms []v1.WeightedPodAffinityTerm) ([]*processedWeightedAffinityTerm, error) {
+	if terms == nil {
+		return nil, nil
 	}
+
+	var processedTerms []*processedWeightedAffinityTerm
+	for i := range terms {
+		p, err := getProcessedWeightedAffinityTerm(pod, &terms[i].PodAffinityTerm, terms[i].Weight)
+		if err != nil {
+			return nil, err
+		}
+		processedTerms = append(processedTerms, p)
+	}
+	return processedTerms, nil
+}
+
+func (p *podAffinityPriorityMap) processTerm(term *processedWeightedAffinityTerm, podDefiningAffinityTerm, podToCheck *v1.Pod, fixedNode *v1.Node, weight int64) error {
 	if len(fixedNode.Labels) == 0 {
 		return nil
 	}
 
-	match := priorityutil.PodMatchesTermsNamespaceAndSelector(podToCheck, namespaces, selector)
-	tpValue, tpValueExist := fixedNode.Labels[term.TopologyKey]
+	match := priorityutil.PodMatchesTermsNamespaceAndSelector(podToCheck, term.namespaces, term.selector)
+	tpValue, tpValueExist := fixedNode.Labels[term.topologyKey]
 	if match && tpValueExist {
 		p.Lock()
-		if p.topologyScore[term.TopologyKey] == nil {
-			p.topologyScore[term.TopologyKey] = make(map[string]int64)
+		if p.topologyScore[term.topologyKey] == nil {
+			p.topologyScore[term.topologyKey] = make(map[string]int64)
 		}
-		p.topologyScore[term.TopologyKey][tpValue] += weight
+		p.topologyScore[term.topologyKey][tpValue] += weight
 		p.Unlock()
 	}
 	return nil
 }
 
-func (p *podAffinityPriorityMap) processTerms(terms []v1.WeightedPodAffinityTerm, podDefiningAffinityTerm, podToCheck *v1.Pod, fixedNode *v1.Node, multiplier int) error {
-	for i := range terms {
-		term := &terms[i]
-		if err := p.processTerm(&term.PodAffinityTerm, podDefiningAffinityTerm, podToCheck, fixedNode, int64(term.Weight*int32(multiplier))); err != nil {
+func (p *podAffinityPriorityMap) processTerms(terms []*processedWeightedAffinityTerm, podDefiningAffinityTerm, podToCheck *v1.Pod, fixedNode *v1.Node, multiplier int) error {
+	for _, term := range terms {
+		if err := p.processTerm(term, podDefiningAffinityTerm, podToCheck, fixedNode, int64(term.weight*int32(multiplier))); err != nil {
```

Comment:
passing the multiplier seems more readable to me.

Can we rename to `updateScoreWithTerms` or something like that?

Reply:
> passing the multiplier seems more readable to me.

done.

> Can we rename to updateScoreWithTerms or something like that?

then we need to come up with a name for "processTerms" as well, let keep like this.